lib/spot-checker.js

```diff
@@ -86,9 +96,12 @@ export default class SpotChecker {
    * @param {string} args.address
    * @param {string} args.cid
    * @param {object} args.stats
-   * @param {number} [args.maxByteLength]
+   * @param {'all' | 'block'} args.dagScope
+   * @param {number} [args.maxByteLength=0] - Maximum bytes to download before stopping retrieval (applies to full retrievals only)
    * */
-  async fetchCAR ({ protocol, address, cid, stats, maxByteLength }) {
+  async fetchCAR ({ protocol, address, cid, stats, dagScope, maxByteLength = 0 }) {
+    const retrievalStats = dagScope === 'all' ? stats.fullRetrieval : stats.blockRetrieval
+    maxByteLength = dagScope === 'all' ? maxByteLength : 0
```

Comment:
Similarly here - let's move this logic outside of `fetchCAR`.

Also let's use `maxByteLength===undefined` to decide whether the `maxByteLength` limit applies. The value `maxByteLength=0` is a valid limit we could use e.g. to simulate a HEAD request.

Reply:
Addressed in https://github.com/CheckerNetwork/spark-spot-check/commit/b5068e3e28f1e841701b98921cf48ef9d2a46b5a